notebooks/122-quantizing-model-with-accuracy-control/122-yolov8-quantization-with-accuracy-control.ipynb

```diff
@@ -0,0 +1,332 @@
+{
```

Comment:
<p>I got error <code>TypeError: cannot pickle 'Model' object </code>with nncf version git+https://github.com/openvinotoolkit/nncf.git@develop. nncf==2.5.0 work slower, but without fail. Do you know, maybe it is required additional modules to install ?</p><p> </p><p><code>File c:\users\sbalandi\me\work\1\openvino_env\lib\site-packages\nncf\quantization\algorithms\accuracy_control\ranker.py:213, in Ranker._multiprocessing_calculation_ranking_score(self, quantized_model, quantized_model_graph, groups_to_rank, ranking_subset_indices, reference_values_for_each_item)</code></p><p><code>&nbsp;&nbsp;208 for idx, current_group in enumerate(groups_to_rank):</code></p><p><code>&nbsp;&nbsp;209&nbsp;&nbsp;&nbsp;modified_model = revert_operations_to_floating_point_precision(</code></p><p><code>&nbsp;&nbsp;210&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;current_group.operations, current_group.quantizers, quantized_model, quantized_model_graph</code></p><p><code>&nbsp;&nbsp;211&nbsp;&nbsp;&nbsp;)</code></p><p><code>--&gt; 213&nbsp;&nbsp;&nbsp;prepared_model_queue.append(self._algo_backend.prepare_for_inference_async(modified_model))</code></p><p><code>&nbsp;&nbsp;215&nbsp;&nbsp;&nbsp;if idx &gt;= (self._num_processes - 1):</code></p><p><code>&nbsp;&nbsp;216&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;prepared_model = prepared_model_queue.pop(0).get()</code></p><p><code>File c:\users\sbalandi\me\work\1\openvino_env\lib\site-packages\nncf\quantization\algorithms\accuracy_control\openvino_backend.py:123, in OVAccuracyControlAlgoBackend.prepare_for_inference_async(model)</code></p><p><code>&nbsp;&nbsp;121 queue = multiprocessing.Queue()</code></p><p><code>&nbsp;&nbsp;122 p = multiprocessing.Process(target=compile_model, args=(model, queue))</code></p><p><code>--&gt; 123 p.start()</code></p><p><code>&nbsp;&nbsp;124 return OVAsyncPreparedModel(p, queue)</code></p><p><code>File ~\AppData\Local\Programs\Python\Python38\lib\multiprocessing\process.py:121, in BaseProcess.start(self)</code></p><p><code>&nbsp;&nbsp;118 assert not _current_process._config.get('daemon'), \</code></p><p><code>&nbsp;&nbsp;119&nbsp;&nbsp;&nbsp;&nbsp;'daemonic processes are not allowed to have children'</code></p><p><code>&nbsp;&nbsp;120 _cleanup()</code></p><p><code>--&gt; 121 self._popen = self._Popen(self)</code></p><p><code>&nbsp;&nbsp;122 self._sentinel = self._popen.sentinel</code></p><p><code>&nbsp;&nbsp;123 # Avoid a refcycle if the target function holds an indirect</code></p><p><code>&nbsp;&nbsp;124 # reference to the process object (see bpo-30775)</code></p><p><code>File ~\AppData\Local\Programs\Python\Python38\lib\multiprocessing\context.py:224, in Process._Popen(process_obj)</code></p><p><code>&nbsp;&nbsp;222 @staticmethod</code></p><p><code>&nbsp;&nbsp;223 def _Popen(process_obj):</code></p><p><code>--&gt; 224&nbsp;&nbsp;&nbsp;return _default_context.get_context().Process._Popen(process_obj)</code></p><p><code>File ~\AppData\Local\Programs\Python\Python38\lib\multiprocessing\context.py:327, in SpawnProcess._Popen(process_obj)</code></p><p><code>&nbsp;&nbsp;324 @staticmethod</code></p><p><code>&nbsp;&nbsp;325 def _Popen(process_obj):</code></p><p><code>&nbsp;&nbsp;326&nbsp;&nbsp;&nbsp;from .popen_spawn_win32 import Popen</code></p><p><code>--&gt; 327&nbsp;&nbsp;&nbsp;return Popen(process_obj)</code></p><p><code>File ~\AppData\Local\Programs\Python\Python38\lib\multiprocessing\popen_spawn_win32.py:93, in Popen.__init__(self, process_obj)</code></p><p><code>&nbsp;&nbsp;&nbsp;91 try:</code></p><p><code>&nbsp;&nbsp;&nbsp;92&nbsp;&nbsp;&nbsp;reduction.dump(prep_data, to_child)</code></p><p><code>---&gt; 93&nbsp;&nbsp;&nbsp;reduction.dump(process_obj, to_child)</code></p><p><code>&nbsp;&nbsp;&nbsp;94 finally:</code></p><p><code>&nbsp;&nbsp;&nbsp;95&nbsp;&nbsp;&nbsp;set_spawning_popen(None)</code></p><p><code>File ~\AppData\Local\Programs\Python\Python38\lib\multiprocessing\reduction.py:60, in dump(obj, file, protocol)</code></p><p><code>&nbsp;&nbsp;&nbsp;58 def dump(obj, file, protocol=None):</code></p><p><code>&nbsp;&nbsp;&nbsp;59&nbsp;&nbsp;&nbsp;'''Replacement for pickle.dump() using ForkingPickler.'''</code></p><p><code>---&gt; 60&nbsp;&nbsp;&nbsp;ForkingPickler(file, protocol).dump(obj)</code></p><p><code>TypeError: cannot pickle 'Model' object</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1234/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Reply:
I tried to reproduce it in a new environment with python3.8 and python3.10  and ubuntu 22.04.  There are no errors. Looks like a bug in nncf in a specific environment.

Reply:
@aleksandr-mokrov yes, you are right, it is bug - CVS-118321 on win, work around for windows now is adding parameter `num_ranking_processes=1` to advanced_accuracy_restorer_parameters (https://github.com/openvinotoolkit/nncf/blob/develop/nncf/quantization/advanced_parameters.py#L201) . Could you please add it ?

Reply:
Added